mpl_data_cast/gui/main.py

```diff
@@ -0,0 +1,235 @@
+        dlg.setWindowTitle("MPL-Data-Cast Preferences")
+        dlg.exec()
+        # update maximum tree depth if necessary
+        if self.settings.value("rtdc/tree_depth_limit",
+                               8) != prev_tree_depth_limit:
+            self.widget_output.tree_depth_limit = int(self.settings.value(
+                "rtdc/tree_depth_limit", 8))
+            self.widget_input.tree_depth_limit = int(self.settings.value(
+                "rtdc/tree_depth_limit", 8))
+            if self.widget_output.path is not None:
+                self.widget_output.update_output_dir(self.widget_output.path)
+            if self.widget_input.path is not None:
+                self.widget_input.update_input_dir(self.widget_input.path)
+
+    @QtCore.pyqtSlot()
+    def on_action_quit(self) -> None:
+        """Determine what happens when the user wants to quit"""
+        QtCore.QCoreApplication.quit()
+
+    def on_action_about(self) -> None:
+        """Show imprint."""
+        gh = "GuckLab/MPL-Data-Cast"
+        rtd = "mpl-data-cast.readthedocs.io"
+        about_text = "Convert and transfer data from measurement PCs to " \
+                     "network shares at MPL.<br><br>" \
+                     + "Author: Paul Müller and others<br>" \
+                     + "GitHub: " \
+                     + "<a href='https://github.com/{gh}'>{gh}</a><br>".format(gh=gh) \
+                     + "Documentation: " \
+                     + "<a href='https://{rtd}'>{rtd}</a><br>".format(rtd=rtd)  # noqa 501
+        QtWidgets.QMessageBox.about(self,
+                                    "MPL-Data-Cast {}".format(version),
+                                    about_text)
+
+    @QtCore.pyqtSlot()
+    def on_action_software(self) -> None:
+        """Show used software packages and dependencies."""
+        libs = [dclab,
+                h5py,
+                numpy,
+                ]
+
+        sw_text = f"MPL-Data-Cast {version}\n\n"
+        sw_text += f"Python {sys.version}\n\n"
+        sw_text += "Modules:\n"
+        for lib in libs:
+            sw_text += f"- {lib.__name__} {lib.__version__}\n"
+        sw_text += f"- PyQt6 {QtCore.QT_VERSION_STR}\n"
+
+        QtWidgets.QMessageBox.information(self, "Software", sw_text)
+
+    @QtCore.pyqtSlot()
+    def on_task_transfer(self) -> None:
+        """Execute recipe to transfer data."""
+        if not self.widget_input.path.exists():
+            QtWidgets.QMessageBox.information(self, "Error",
+                                              "Input directory not correct!")
+        if not self.widget_output.path.exists():
+            QtWidgets.QMessageBox.information(self, "Error",
+                                              "Output directory not correct!")
+
+        rp = RTDCRecipe(self.widget_input.path, self.widget_output.path)
+
+        nb_files = 0  # counter for files, used for progress bar
+        for elem in self.widget_input.path.rglob("*.*"):
```

Comment:
To get the correct number of files to transfer, you should probably use `len(list(rp.get_raw_data_iterator))`, or am I missing something?
The way you are doing it here, you are taking into account **all** files, even those not being tranferred (e.g. Settings.ini and such).

Reply:
The software will copy all files since that behavior is defined in the class `Recipe`. It's just that for the recipe "RTDC", the "settings.ini" files are processed such that they are stored in the logs of the corresponding .rtdc file.  
To ignore files that are not copied, I will add another condition to that "if" statement that checks if the filename appears in the list `IGNORED_FILE_NAMES` that is defined in `recipe.py`. That should cover most scenarios and be accurate enough for users.

What do you think about adding a statement like "The progress bar will indicate how many files have already been processed." to the docs?

Reply:
I still don't believe this is right and you should use ``get_raw_data_iterator`` here. In any case, are files named ``I-have-no-extension`` covered by this?

Reply:
Files without extension are not covered by this yet, I will change that (glob for "*" instead of "*.*").

I tested it now with `get_raw_data_iterator` and how I implemented it. I created a folder with in total ten files, five of them .rtdc files, then one .txt file and four PDFs. With the data iterator function, there are five files detected and the progress bar jumps in steps of 20%. All files are copied.
glob-ing files and ignoring known junk files, the code correctly finds ten files and the progress bar jumps in steps of 10%. All files are copied.

Personally, I like the second approach (how I implemented it now) because it shows the number of files that are actually copied, therefore giving a much better estimate of the remaining time for the transfer.